homework7/src/main/java/raisetech/homework7/RestController.java

```diff
@@ -37,10 +37,9 @@ public ResponseEntity<String> createUser(@RequestBody @RequestParam(value = "nam
 
   //PATCHメソッドはPOSTと同じく入力した文字列が空文字、null、20文字以上なら受け取らないで、URLパラメータの「names/」の後に整数を入れる仕様
   @PatchMapping("/names/{id}")
-  public ResponseEntity<Map<String, String>> updateUser(@PathVariable("id") int id, @RequestBody @RequestParam(value = "name") String name, @Validated UserForm form, UriComponentsBuilder builder) {
-    String idNumber = String.valueOf(id);
+  public ResponseEntity<Map<String, String>> updateUser(@PathVariable("id") String id, @RequestBody @RequestParam(value = "name") String name, @Validated UserForm form, UriComponentsBuilder builder) {
 
-    String path = "/names/" + idNumber + form.getName();
+    String path = "/names/" + form.getId() + form.getName();

```

Comment:
formからidは入力されないと思うので、パス変数からidを取得する必要があります！

Reply:
パス変数からidを取得するというのがよく分からないです。
後、一応通信はできています。
![image](https://user-images.githubusercontent.com/111167638/202075722-2e3421b2-c5e7-4628-95c1-3c3a99364979.png)
![image](https://user-images.githubusercontent.com/111167638/202075797-9de19cd9-c2e9-4d4d-81b8-f3011f4de85b.png)

Reply:
@kaidou2
> パス変数からidを取得するというのがよく分からない

以前の実装でやっていたことだと思うのですが、どこが分からないか具体的に教えて頂けないですか・・・？
 
```suggestion
   String id = String.valueOf(id);
```

Reply:
パス変数というのはpathのことでしょうか？
> formからidは入力されないと思う
PATCHメソッドの引数にidがあって、userFormクラスに「private String id」があるので値が渡っていると思ったのですが、違うのでしょうか？間違ってるとしたらidはnameと違って@PatchMapping("/names/{id}")に{}で囲んでいることや、@RequestParam(value = "name")じゃなくて@PathVariable("id")で指定していることに原因があるのかもしれないとか思ってます。

Reply:
> パス変数というのはpathのことでしょうか？

違いますね！
調べてみてください〜